src/seedu/addressbook/data/person/Name.java

```diff
@@ -2,63 +2,95 @@
 
 import seedu.addressbook.data.exception.IllegalValueException;
 
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 
 /**
- * Represents a Person's name in the address book.
- * Guarantees: immutable; is valid as declared in {@link #isValidName(String)}
+ * Represents a Person's name in the address book. Guarantees: immutable; is
+ * valid as declared in {@link #isValidName(String)}
  */
 public class Name {
 
-    public static final String EXAMPLE = "John Doe";
-    public static final String MESSAGE_NAME_CONSTRAINTS = "Person names should be spaces or alphabetic characters";
-    public static final String NAME_VALIDATION_REGEX = "[\\p{Alpha} ]+";
+	public static final String EXAMPLE = "John Doe";
+	public static final String MESSAGE_NAME_CONSTRAINTS = "Person names should be spaces or alphabetic characters";
+	public static final String NAME_VALIDATION_REGEX = "[\\p{Alpha} ]+";
 
-    public final String fullName;
+	public final String fullName;
 
-    /**
-     * Validates given name.
-     *
-     * @throws IllegalValueException if given name string is invalid.
-     */
-    public Name(String name) throws IllegalValueException {
-        name = name.trim();
-        if (!isValidName(name)) {
-            throw new IllegalValueException(MESSAGE_NAME_CONSTRAINTS);
-        }
-        this.fullName = name;
-    }
+	/**
+	 * Validates given name.
+	 *
+	 * @throws IllegalValueException
+	 *             if given name string is invalid.
+	 */
+	public Name(String name) throws IllegalValueException {
+		name = name.trim();
```

Comment:
Indentation problem here?. Our coding standard requires you to use 4 spaces instead of tabs. You can configure Eclipse to convert tabs to spaces automatically.